modules/uipath/src/backend/index.ts

```diff
@@ -0,0 +1,104 @@
+import * as sdk from 'botpress/sdk'
+import jsonwebtoken from 'jsonwebtoken'
+
+export class ResponseError extends Error {
+  errorCode: string
+  statusCode: number
+
+  skipLogging = false
+
+  constructor(message: string, statusCode: number, errorCode: string) {
+    super(message)
+    Error.captureStackTrace(this, this.constructor)
+    this.statusCode = statusCode
+    this.errorCode = errorCode
+  }
+}
+
+class UnauthorizedError extends ResponseError {
+  type = 'UnauthorizedError'
+
+  constructor(message: string) {
+    super(`Unauthorized: ${message}`, 401, 'BP_0041')
+  }
+}
+
+const onServerStarted = async (bp: typeof sdk) => {
+  const authRouter = bp.http.createRouterForBot('uipath/auth', {
+    checkAuthentication: true
+  })
+
+  const rootRouter = bp.http.createRouterForBot('uipath', {
+    checkAuthentication: false
+  })
+
+  authRouter.get('/token', async (req, res) => {
+    const expiresIn = req.query.expiresIn || '1m'
+
+    try {
+      const token = jsonwebtoken.sign({ text: 'something' }, process.APP_SECRET, { expiresIn })
+      res.status(200).send({ token })
+    } catch (err) {
+      res.status(400).send(err)
+    }
+  })
+
+  rootRouter.post(
+    '/message',
+    async (req, res, next) => {
+      if (!req.headers.authorization) {
```

Comment:
If you think of adding more routes maybe extracting this as a middleware would make sens

Reply:
Will do once I add more routes

Reply:
I really think it should be moved to a middleware, even if it's only for one route. Keeps the code more clear, and separate code concerns.